src/Swift.Bindings/src/Emitter/StringEmitter/Handler/MethodHandler.cs

```diff
@@ -318,14 +319,17 @@ public void HandleReturnType()
                 return;
             }
 
-            if (_env.MethodDecl.IsAsync && !MarshallingHelpers.ArgumentIsMarshalledAsCSStruct(returnType, _env.TypeDatabase))
+            TypeRecord returnTypeRecord = _env.TypeDatabase.GetTypeRecordOrThrow(returnType.SwiftTypeSpec);
+            if (_env.MethodDecl.IsAsync && !MarshallingHelpers.IsTypeFrozen(returnTypeRecord))
             {
                 SetReturnType("IntPtr");
                 return;
             }
 
-            var returnTypeRecord = _env.TypeDatabase.GetTypeRecordOrThrow(returnType.SwiftTypeSpec);
-            SetReturnType(returnTypeRecord.CSharpTypeName.FullyQualifiedName);
+            if (MarshallingHelpers.IsTypeHeapAllocated(returnTypeRecord))
+                SetReturnType(returnTypeRecord.CSharpTypeName.FullyQualifiedName + ".Buffer");
```

Comment:
Can csharp types contain "."?

Reply:
It isn’t allowed as part of an individual type’s identifier. Here, "." is used as a delimiter for the nested buffer type.

Reply:
Consider the following scenario where a frozen Swift struct is projected as a C# class due to memory handling requirements. In Swift, we have:
```swift
public class RefType {
}

@frozen
public struct FrozenStructHeapAllocated {
    public var a: RefType
    public var b: Int32
}
```
This struct is projected into C# as:
```csharp
public class FrozenStructHeapAllocated {
  public struct Buffer {
     IntPtr _a;
     Int _b;
  }
  
  ~FrozenStructHeapAllocated() {
    // VWT->Destroy
  }
}
```

At the ABI boundary, the struct is enregistered as `FrozenStructHeapAllocated.Buffer`.

Reply:
I see, thanks for explanation!